### DISCUSSION TITLE
Algorithms

### DISCUSSION THREAD

**Discussion by Ugurmutturkel** [ahmedIbrahim84201/AlgoLingo_Senior_Project#6] [General]
Which algorithms should we implement in the project? can we get help from a teacher or a professional to explain them more thoroughly?

**Comment by VegaCaroMiguel** [comment]
I think the best you can do is to create a curated list of topics, from basics to easy-medium-hard and a curated list for each difficulty on each data structure, I think the best teacher to explain data structures is to solve lots problems from super-naive  difficulty to hard difficulty.
Sites like Codeforces, Atcoder and Hackerrank are good project bases. 

I think that if you want a gamified experience, contests like the ones in CodeForces would be a great fit, here are the basic data structures:
![image](https://github.com/ahmedIbrahim84201/AlgoLingo_Senior_Project/assets/69682287/a0990ba6-e9e1-4cb8-85dc-9fcf762d42b5)

Regarding the algos, I think cp algortithms set is one of the best curated lists of algorithms problems in the internet:
https://cp-algorithms.com/index.html

CSES is also a good resource regarding to algorithms problems.
https://cses.fi/problemset/

USACO guide is the platforms where us highschools students train for the IOI(International Olympiad of Informatics), very cool UX as well.


I hope that helps,
Have a nice one !

**Reply by Ugurmutturkel** [reply]
Thank you! we will consider your comments with our project
